site/profile/manifests/core/icinga_agent.pp

```diff
@@ -0,0 +1,94 @@
+# @summary
+#   Icinga agent creation for metric collections
+
+class profile::core::icinga_agent(
+  String $icinga_master_fqdn,
+  String $icinga_master_ip,
+  String $credentials_hash,
+  String $host_template,
+){
+  $packages = [
+    'nagios-plugins-all',
+  ]
+  $icinga_agent_fqdn = $facts['networking']['fqdn']
+  $icinga_agent_ip = $facts['networking']['ip']
+  $credentials = "Authorization:Basic ${credentials_hash}"
+  $json_file = "{
+\"address\": \"${icinga_agent_ip}\",
+\"display_name\": \"${icinga_agent_fqdn}\",
+\"imports\": [
+  \"${host_template}\"
+],
+\"object_name\":\"${icinga_agent_fqdn}\",
+\"object_type\": \"object\",
+\"vars\": {
+    \"safed_profile\": \"3\"
+}
+}"
+  $icinga_path = '/opt/icinga'
+  $path = "${icinga_path}/${icinga_agent_fqdn}.json"
+  $url = "https://${icinga_master_fqdn}/director/host"
+  $cmd = "curl -s -k -H '${credentials}' -H 'Accept: application/json' -X POST '${url}' -d @${path}"
+  $cond = "curl -s -k -H '${credentials}' -H 'Accept: application/json' -X GET '${url}/host?name=${icinga_agent_fqdn}' | grep Failed"
+
+##Create a directory to allocate json files
```

Comment:
Comments need to be indented to the same depth as the code they are referring too.

Reply:
all comments indented.